scripts/configurator/parser/cmake_row.py

```diff
@@ -87,18 +62,16 @@ def is_pound_comment(self, row: str) -> bool:
       return row[0] == "#"
 
    def serialize(self) -> str:
-      serialized_val: str
       if self.row_type == row_type.VARIABLE:
-         if self.configurator_type == configurator_type.BOOL:
-            serialized_val = "ON" if self.val else "OFF"
-         else:
-            serialized_val = str(self.val)
-         return self.name + ":" + self._cmake_type_str + "="  + serialized_val
-      elif self.row_type == row_type.POUND_COMMENT:
+         return self.name + ":" + self._cmake_type + "=" + self.val.serialize()
+      elif type(self.val) == str and self.row_type == row_type.POUND_COMMENT:
          return "#" + self.val
-      elif self.row_type == row_type.SLASH_COMMENT:
+      elif type(self.val) == str and self.row_type == row_type.SLASH_COMMENT:
```

Comment:
Same as before.

NIT: It would be more polished to have an `elif` case for the empty string and a final `else` statement where you throw an exception. In other words, the `if/elif` list here must be _exhaustive_ and cover all the types that `row` could possibly have. In the current implementation, you simply assume that there are no more types to cover and default to empty string case: that is correct (obviously) in this case, but "doesn't scale" in large projects with many types to cover. The cool thing about C and C++ switch statements is that, if you do it on an enum type, the compiler will show you a warning in case you missed a case. Anyway, this is an optional NIT as we don't have that scale here and `row` won't have many more types.

Reply:
Ok, this makes sense